crates/bevy_ecs/src/schedule/executor/single_threaded.rs

```diff
@@ -23,6 +23,23 @@ pub struct SingleThreadedExecutor {
     unapplied_systems: FixedBitSet,
     /// Setting when true applies system buffers after all systems have run
     apply_final_buffers: bool,
+
+    /// Set to true enable system stepping mode
+    step_state: StepState,
+}
+
+/// step type when stepping is enabled
+#[derive(Default, Copy, Clone, PartialEq, Debug)]
+enum StepState {
+    /// Don't run any systems that respect stepping
+    Wait(usize),
+    /// Only run the first system that hasn't yet been run
+    Next(usize),
+    /// Run all remaining systems
+    Rest(usize),
```

Comment:
Rest is a poor choice of name here, since it can also be a verb that means "wait". Remaining?